### PR TITLE
Added GPT API cost calculation

### PR DESCRIPTION
**Pull request by LinfanS**
Hi,

This pull request for the addition of a quick script to iterate through the lines of the console logs from running the batch pipeline, isolating the list of values related to the token usage for the ChatGPT API and calculates the total cost in USD.

Doesn't account of course for additional cost of testing and developing the pipeline and ~~having to run the batch pipeline again after realising it's not working~~ but a cost figure is a nice to have metric to presenting to stakeholders.

### CODE REVIEW

**Review by annac02 — APPROVED** [review]
I would just add a comment that has like a link to where you got these values from or something like that (or mention it in readme) but nice, good to have and good variable names :)

File: pipeline/calculate_gpt_cost.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        if "GPT-4o-mini usage cost:" in line:
+            cost_string = line.split(": ")[1]
+            cost_list = literal_eval(cost_string)
+            output_tokens += cost_list[0]
+            input_tokens += cost_list[1]
+
+    output_cost = (output_tokens / 1000000) * OUTPUT_COST_PER_MILLION_TOKENS
+    input_cost = (input_tokens / 1000000) * INPUT_COST_PER_MILLION_TOKENS
+    total_cost = output_cost + input_cost
+    total_cost_usd = f"${total_cost:.2f}"
```
Comment (alabbahs) [inline pipeline/calculate_gpt_cost.py:26]: maybe print input and output costs too just for some more insight.

**Review by alabbahs — APPROVED** [review]

**Comment by alabbahs** [comment]
Beautiful idea and cleanly implemented.

File: pipeline/calculate_gpt_cost.py (hunk 1 above)
Comment (LinfanS) [reply pipeline/calculate_gpt_cost.py:29]: Sure :)

**Comment by LinfanS** [comment]
> I would just add a comment that has like a link to where you got these values from or something like that (or mention it in readme) but nice, good to have and good variable names :)

Thanks Anna, I've added this suggestion. Thank you.

Changes have been implemented so merging this to main.

**Pull request merged**